contracts/introspection/ERC165Checker.sol

```diff
@@ -0,0 +1,149 @@
+   * @return true if the contract at _address implements ERC165
+   */
+  function supportsERC165(address _address)
+    internal
+    view
+    returns (bool)
+  {
+    // Any contract that implements ERC165 must explicitly indicate support of
+    // InterfaceId_ERC165 and explicitly indicate non-support of InterfaceId_Invalid
+    return supportsERC165Interface(_address, InterfaceId_ERC165) &&
+      !supportsERC165Interface(_address, InterfaceId_Invalid);
+  }
+
+  /**
+   * @notice Query if a contract implements an interface, also checks support of ERC165
+   * @param _address The address of the contract to query for support of an interface
+   * @param _interfaceId The interface identifier, as specified in ERC-165
+   * @return true if the contract at _address indicates support of the interface with
+   * identifier _interfaceId, false otherwise
+   * @dev Interface identification is specified in ERC-165.
+   */
+  function supportsInterface(address _address, bytes4 _interfaceId)
+    internal
+    view
+    returns (bool)
+  {
+    // query support of both ERC165 as per the spec and support of _interfaceId
+    return supportsERC165(_address) &&
+      supportsERC165Interface(_address, _interfaceId);
+  }
+
+  /**
+   * @notice Query if a contract implements interfaces, also checks support of ERC165
+   * @param _address The address of the contract to query for support of an interface
+   * @param _interfaceIds A list of interface identifiers, as specified in ERC-165
+   * @return true if the contract at _address indicates support all interfaces in the
+   * _interfaceIds list, false otherwise
+   * @dev Interface identification is specified in ERC-165.
+   */
+  function supportsInterfaces(address _address, bytes4[] _interfaceIds)
+    internal
+    view
+    returns (bool)
+  {
+    // query support of ERC165 itself
+    if (!supportsERC165(_address)) {
+      return false;
+    }
+
+    // query support of each interface in _interfaceIds
+    for (uint256 i = 0; i < _interfaceIds.length; i++) {
+      if (!supportsERC165Interface(_address, _interfaceIds[i])) {
+        return false;
+      }
+    }
+
+    // all interfaces supported
+    return true;
+  }
+
+  /**
+   * @notice Query if a contract implements an interface, does not check ERC165 support
+   * @param _address The address of the contract to query for support of an interface
+   * @param _interfaceId The interface identifier, as specified in ERC-165
+   * @return true if the contract at _address indicates support of the interface with
+   * identifier _interfaceId, false otherwise
+   * @dev Assumes that _address contains a contract that supports ERC165, otherwise
+   * the behavior of this method is undefined. This precondition can be checked
+   * with the `supportsERC165` method in this library.
+   * Interface identification is specified in ERC-165.
+   */
+  function supportsERC165Interface(address _address, bytes4 _interfaceId)
+    internal
```

Comment:
I had changed this to `private` as per @frangio's request: https://github.com/OpenZeppelin/openzeppelin-solidity/pull/1086#discussion_r210733067

Since we now have the `supportsInterfaces` API to help save gas when checking many interfaces, we should make `supportsERC165Interface` private.

Let me know if you agree and I can make this change.